ggshield/verticals/sca/file_selection.py

```diff
@@ -99,6 +99,8 @@ def sca_files_from_git_repo(
     if isinstance(sca_files_result, Detail):
         if sca_files_result.status_code == 401:
             raise APIKeyCheckError(client.base_uri, "Invalid API key.")
+        elif sca_files_result.status_code == 501:
+            raise UnexpectedError(sca_files_result.detail)
```

Comment:
Should we define a specific error in this case ?

Reply:
I don't know, actually in other parts of the code (e.g. here https://github.com/GitGuardian/ggshield/blob/main/ggshield/cmd/sca/scan/sca_scan_utils.py#L240) we always display the content of any unexpected error. Maybe we should also do that here? (I didn't want to touch to existing code but maybe it's OK?)

Reply:
My point was just to not use `UnexpectedError`, from a semantic point of view. Maybe a `FeatureUnavailable` or something like that.